compiler/packages/react-mcp-server/src/utils/runtimePerf.ts

```diff
@@ -0,0 +1,138 @@
+        plugins: [
+            'jsx',
+            'typescript',
+            'classProperties',
+            'optionalChaining',
+            'nullishCoalescingOperator',
+        ],
+    });
+}
+
+/**
+ * Transform AST into browser-compatible JavaScript
+ * @param {babel.types.File} ast - The AST to transform
+ * @param {Object} opts - Transformation options
+ * @returns {Promise<string>} - The transpiled code
+ */
+async function transformAsync(ast: babel.types.Node, code: string) {
+    // Provide a dummy filename to satisfy Babel's requirement for filename context
+    const result = await babel.transformFromAstAsync(ast, null,  {
+        filename: 'file.jsx',
+        presets: [
+          ['@babel/preset-env'],
+          '@babel/preset-react'
+        ],
+        plugins: [
+            () => ({
+                visitor: {
+                  ImportDeclaration(path) {
+                    const value = path.node.source.value;
+                    if (value === 'react' || value === 'react-dom') {
+                      path.remove();
+                    }
+                  }
+                }
+              })
+        ]
+    });
+
+    return result?.code || '';
+}
+
+function buildHtml(transpiled: string | null | undefined) {
+    // Create HTML that includes React, ReactDOM, and the transpiled code
+    const html = `
+        <!DOCTYPE html>
+        <html>
+        <head>
+            <meta charset="UTF-8">
+            <title>React Performance Test</title>
+            <script crossorigin src="https://unpkg.com/react@18/umd/react.development.js"></script>
+            <script crossorigin src="https://unpkg.com/react-dom@18/umd/react-dom.development.js"></script>
+            <script src="https://unpkg.com/web-vitals@3.0.0/dist/web-vitals.iife.js"></script>
+            <style>
+                body { margin: 0; font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif; }
+                #root { padding: 20px; }
+            </style>
+        </head>
+        <body>
+            <div id="root"></div>
+            <script type="module">
+                // Store performance metrics
+                window.__RESULT__ = {
+                    renderTime: null,
+                    webVitals: {}
+                };
+
+                // Measure web vitals
+                webVitals.onCLS((metric) => { window.__RESULT__.webVitals.cls = metric; });
+                webVitals.onLCP((metric) => { window.__RESULT__.webVitals.lcp = metric; });
+                webVitals.onINP((metric) => { window.__RESULT__.webVitals.inp = metric; });
+                webVitals.onFID((metric) => { window.__RESULT__.webVitals.fid = metric; });
+                webVitals.onTTFB((metric) => { window.__RESULT__.webVitals.ttfb = metric; });
+
+                // Wrap user code in React.Profiler to measure render performance
+                const renderStart = performance.now();
```

Comment:
where's the `renderEnd`?

Reply:
also this seems unnecessary given that React.Profiler already gives you the time taken to mount.